packages/community-cli-plugin/src/utils/loadMetroConfig.js

```diff
@@ -71,6 +77,12 @@ function getOverrideConfig(
       ],
     },
   };
+
+  if (config.watchFolders.length === 0) {
+    overrides.watchFolders = getWatchFolders(ctx.root);
+  }
```

Comment:
The only problem I see with this approach of providing the default as an override after loading the user-land config, is that the developer has to provide at least one element in the `watchFolders` to disable this default.

Reply:
Another alternative would be to update the `metro-config` package to not provide an empty array for `watchFolders` and match on `undefined` instead of an empty array here. But .. that might be slightly breaking to developers that use the default `watchFolders` array by pushing their values into it 🤔